Source/ReactiveCore/MoyaError.swift

```diff
@@ -9,3 +9,59 @@ public enum MoyaErrorCode: Int {
     case StatusCode
     case Data
 }
+
+public enum MoyaError: ErrorType {
+    case ImageMapping(MoyaResponse)
+    case JSONMapping(MoyaResponse)
+    case StringMapping(MoyaResponse)
+    case StatusCode(Int,MoyaResponse)
+    case Data(MoyaResponse)
+    case Underlying(ErrorType)
+    
+    public func sameErrorType(otherError: MoyaError) -> Bool {
```

Comment:
Agreed, this function works great. Moving it to the test target sounds great too, we can always move it back later.